pyanaconda/ui/gui/spokes/installation_source.py

```diff
@@ -451,7 +460,11 @@ def _update_payload_source(self):
         source_proxy = self.payload.get_source_proxy()
         source_type = source_proxy.Type
 
-        if self._autodetect_button.get_active():
+        if self._cdn_button.get_active():
+            if source_type == SOURCE_TYPE_CDN:
+                return False
+            switch_source(self.payload, SOURCE_TYPE_CDN)
```

Comment:
Please, extend the SourceSwitchHandler class.

Reply:
We need the `switch_source()` for use in the subscription handling thread, so I wanted to avoid code duplication by using that.